src/topkg_jbuilder.ml

```diff
@@ -1,11 +1,57 @@
 open StdLabels
 open Topkg
+open Fpath
+
+let uerror = Unix.error_message
+
+let rec mkdir dir =
+  let mode = 0o755 in
+  let aux d =
+    Log.debug (fun l -> l "mkdir %s" dir);
+    try Ok (Unix.mkdir d mode) with
+    | Unix.Unix_error (Unix.EEXIST, _, _) -> Ok ()
+    | Unix.Unix_error (e, _, _) ->
+      if d = dir
+      then R.error_msgf "create directory %s: %s" d (uerror e)
+      else R.error_msgf "create directory %s: %s: %s" dir d (uerror e)
+  in
+  OS.Dir.exists dir >>= function
+  | true  -> Ok ()
+  | false ->
+    let parent = Fpath.dirname dir in
+    OS.Dir.exists parent >>= function
+    | true  -> aux dir
+    | false -> mkdir parent >>= fun () -> aux dir
+
+let rec copy src dst =
+  OS.Dir.contents ~rel:true src >>= fun paths ->
+  OS.File.fold (fun path acc ->
+      acc >>= fun () ->
+      OS.File.exists (src // path) >>= function
+      | false -> copy (src // path) (dst // path)
+      | true  ->
+        Log.debug
+          (fun l -> l "copy %s/%s to %s/%s" src path dst path);
+        OS.File.read (src // path) >>= fun file ->
+        mkdir @@ Filename.dirname (dst // path) >>= fun () ->
+        OS.File.write (dst // path) file
+    ) (Ok ()) paths
+  >>= fun acc -> acc
 
 let build =
   Pkg.build ()
-    ~cmd:(fun _c os _files ->
+    ~cmd:(fun _c os files ->
       let jbuilder = Conf.tool "jbuilder" os in
-      OS.Cmd.run @@ Cmd.(jbuilder % "build" % "--root" % "."))
+      Log.debug (fun l -> l "files=%s" (String.concat ", " files));
+      match List.rev files with
+      | "doc/api.docdir/index.html" :: _ -> (* topkg doc *)
```

Comment:
Last time I looked at the topkg API I didn't see anything about it. But I think @dbuenzli wants to rework the handling of documentation in topkg.

In the meantime, let's just add a comment explaining what's happening. We can probably use `List.mem` as well rather than checking only the last target.